gradefetcher/gradefetcher.py

```diff
@@ -197,6 +216,26 @@ def student_view(self, context=None):
         frag.add_javascript(self.load_resource("static/js/src/gradefetcher.js"))
         frag.initialize_js("GradeFetcherXBlock")
         return frag
+
+    def studio_view(self, context=None):
+        fragment = Fragment()
+        context = {'fields': []}
+        # Build a list of all the fields that can be edited:
+        for field_name in self.editable_fields:
+            field = self.fields[field_name]
+            assert field.scope in (Scope.content, Scope.settings), (
```

Comment:
Beware that if you run python with the `-O` flag, it doesn't execute `assert` statements. I assume that edX doesn't use that flag, but especially since this is in a reusable module and there isn't any way to control what flags get used, it's risky to rely on `assert` to avoid bad data or enforce anything like a security constraint.

Reply:
oh man, I didn't know that. better clean up the `assert`s in our code.

thanks @thraxil

Reply:
Bandit actually flags it as a security issue, which is one of the reasons that it's handy to have set up on our repos.

The history behind it is that in C, `assert` is usually stripped out of executables by default (you have to enable it for debugging builds), So C programmers get in the habit of putting `assert some_really_expensive_check()` type tests in their code and expect to have them removed from production. Python has them enabled by default, but since Python itself is written by C programmers, they carried over the expectation of stripping them out from "optimized" environments.